Modules/_io/textio.c

```diff
@@ -3173,7 +3175,7 @@ _io_TextIOWrapper_close_impl(textio *self)
 }
 
 static PyObject *
-textiowrapper_iternext(PyObject *op)
+textiowrapper_iternext_locked(PyObject *op)
 {
```

Comment:
`lock_held` is the general convention, and we should add an assertion:
```suggestion
textiowrapper_iternext_lock_held(PyObject *op)
{
    _Py_CRITICAL_SECTION_ASSERT_OBJECT_LOCKED(op);
```